ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -102,6 +105,21 @@ let send_diagnostics rpc doc =
 
 let on_initialize _rpc state _params = Ok (state, initializeInfo)
 
+let code_action_of_case_analysis uri (loc, newText) =
+  let edit : Lsp.Protocol.WorkspaceEdit.t =
+    let textedit : Lsp.Protocol.TextEdit.t =
+      { range = range_of_loc loc; newText }
+    in
+    { changes = Some [ (uri, [ textedit ]) ]; documentChanges = None }
+  in
+  let title = "destruct" in
```

Comment:
nit: not sure this is the best name for this action, at least we could capitalize it